orchagent/portsorch.cpp

```diff
@@ -345,6 +350,132 @@ bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
     return true;
 }
 
+bool PortsOrch::setPortPvid (Port &port, sai_uint32_t pvid)
+{
+    SWSS_LOG_ENTER();
+
+    vector<sai_object_id_t>   idv;
+
+    if (port.m_type == Port::PHY)
+    {
+        idv.push_back(port.m_port_id);
+    }
+    else if (port.m_type == Port::LAG)
+    {
+        Port member;
+        auto it = port.m_members.begin();
+         while (it != port.m_members.end())
+         {
+            const string& memberName = *it;
+            if (!getPort(memberName, member))
+            {
+                SWSS_LOG_ERROR("Failed to get port for %s alias\n", memberName.c_str());
+                return false;
+            }
+            idv.push_back(member.m_port_id);
+            it++;
+         }
+    }
+    else
+    {
+        SWSS_LOG_ERROR("PortsOrch::setPortPvid port type %d not supported", port.m_type);
+        return false;
+    }
+
+    for (const auto& id: idv)
+    {
+        sai_attribute_t attr;
+        attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+        attr.value.u32 = pvid;
+
+        sai_status_t status = sai_port_api->set_port_attribute(id, &attr);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set pvid %u to port pid:%lx", attr.value.u32, id);
+            return false;
+        }
+        SWSS_LOG_INFO("Set pvid %u to port pid:%lx", attr.value.u32, id);
+    }
+    return true;
+}
+
+bool PortsOrch::getPortPvid(Port &port, sai_uint32_t &pvid)
+{
+
+    if (port.m_port_vlan_id != DEFAULT_PORT_VLAN_ID)
+    {
+        pvid = port.m_port_vlan_id;
+        return true;
+    }
+
+    if (port.m_vlan_members.size() == 0)
+    {
+        pvid = DEFAULT_PORT_VLAN_ID;
+        return true;
+    }
+
+    for (auto &vme: port.m_vlan_members)
+    {
+        if(vme.second.vlan_mode == SAI_VLAN_TAGGING_MODE_UNTAGGED) {
+            pvid =  vme.first;
+            return true;
+        }
+    }
+    SWSS_LOG_ERROR("Failed to get pvid for port %s",  port.m_alias.c_str());
+    return false;
+}
+bool PortsOrch::setHostIntfsStripTag(Port &port, sai_hostif_vlan_tag_t strip)
+{
+    SWSS_LOG_ENTER();
+
+    vector<Port *>   portv;
+
+    if (port.m_type == Port::PHY)
+    {
+        portv.push_back(&port);
+    }
+    else if (port.m_type == Port::LAG)
```

Comment:
I think we could add a function called `getLagMembers` to return a vector of `Port`.

Reply:
ok.